Nl symmetries

attacc opened the pull request:
This branch implements very small changes in the NL part:

**Bugs:**
- Fixed bug in output angle loop in angles in yambo_nl
- Fixed bug with reading ndb.QP in yambo_nl
- Fixed bug with collisions compression that was always zero

**Additions:**
 - ypp now save the full symmetries in a new database ndb.full_SYMM when symmetries are removed
 - Possibility to use full symmetries to symmetrize dipoles and residual symmetries for current/polarization
  (not a great improvement, false by default)
 - current operator extended to 4nd order in dk
- added possibility to defined max-rotation angle in yambo_nl

**Changes:**
- Loop with many cgemv transformed in a single cgemm in yambo_nl (suggestion by Andrea F.)
-  changed notation of WF_shifts from 2,1,0 to -1,0,1 according to Davide's suggestion
- W(E)E operator split in  W(E)=W(0)*E + [W(E)-W(0)]*E

attacc committed:
Version 5.3.0,  Revision 23928,  Hash 06985e91e

MODIFIED *  configure include/version/version.m4 hamiltonian/Berry_polarization_NEQ.F nloptics/NL_Berry_current.F

Bugs:
-

Additions:
 -Current and Polarization are symmetrized using residual symmetries

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23929,  Hash ffad4fbd8

MODIFIED *  configure include/version/version.m4 modules/mod_R_lattice.F

Bugs:
-

Additions:
 -New veriable for the small-symmetries group of each k-point

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23930,  Hash cad1cb26f

MODIFIED *  configure include/version/version.m4 bz_ops/.objects

NEW *  bz_ops/k_small_symmetry_group.F

Bugs:
-

Additions:
 -New subroutine to find the small symmetry group of each k-point

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23931,  Hash 3de108411

MODIFIED *  configure include/version/version.m4 bz_ops/k_small_symmetry_group.F modules/mod_R_lattice.F

Bugs:
-

Additions:
 -Work in progress

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23932,  Hash dcdefaf7c

MODIFIED *  configure driver/yambo.F include/version/version.m4 bz_ops/k_small_symmetry_group.F modules/mod_R_lattice.F

Bugs:
-

Additions:
 -Work in progress

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Work in progress

Version 5.3.0,  Revision 23933,  Hash dd5d293b3

MODIFIED *  configure driver/yambo.F include/version/version.m4

Bugs:
-

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23934,  Hash bfdab75b4

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_W_operator.F

Bugs:
-

Additions:
 -Added symmetrization W operator

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23935,  Hash cfb6f015c

MODIFIED *  configure driver/yambo.F include/version/version.m4 modules/mod_nl_optics.F nloptics/NL_driver.F

Bugs:
-

Additions:
 -Word in programs

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23936,  Hash 180c29e07

MODIFIED *  configure include/version/version.m4 hamiltonian/Berry_polarization_NEQ.F interface/INIT_activate.F interface/INIT_load.F modules/mod_electric.F nloptics/NL_Berry_current.F nloptics/NL_driver.F nloptics/NL_initialize.F

Bugs:
-

Additions:
 -New flag to check the use of residual symmetries

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23937,  Hash aa409f9b2

MODIFIED *  configure include/version/version.m4 bz_ops/k_small_symmetry_group.F modules/mod_nl_optics.F nloptics/NL_driver.F nloptics/NL_initialize.F

Bugs:
-

Additions:
 -More stuff

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23938,  Hash cc02184cf

MODIFIED *  configure include/version/version.m4 bz_ops/k_small_symmetry_group.F

Bugs:
-

Additions:
 -Added small group

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23939,  Hash 00f9d567a

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_W_operator.F

Bugs:
-

Additions:
 -Work in progress

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23940,  Hash 6d538ed50

MODIFIED *  configure include/version/version.m4 hamiltonian/.objects hamiltonian/Build_W_operator.F nloptics/NL_Hamiltonian.F sc/SC_add_efield.F

NEW *  hamiltonian/Apply_W_field.F

Bugs:
-

Additions:
-

Changes:
 -Build_W_operator split in two parts

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23941,  Hash 683fdf3d2

MODIFIED *  configure include/version/version.m4 hamiltonian/Apply_W_field.F

Bugs:
-

Additions:
-

Changes:
 -Removed unused variables

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23942,  Hash 9c8dba2e6

MODIFIED *  configure include/version/version.m4 bz_ops/DOUBLE_project.dep hamiltonian/.objects hamiltonian/DOUBLE_project.dep hamiltonian/SC_project.dep

Bugs:
 -Fixed compilation

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23943,  Hash e81740b4f

MODIFIED *  configure include/version/version.m4 hamiltonian/Apply_W_field.F

Bugs:
 -Fixed typos

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23944,  Hash fca94a650

MODIFIED *  configure include/version/version.m4 hamiltonian/Apply_W_field.F modules/mod_electric.F modules/mod_nl_optics.F nloptics/NL_Hamiltonian.F nloptics/NL_initialize.F sc/SC_add_efield.F

Bugs:
-

Additions:
 -Added symmetrization of Wk operator

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23945,  Hash 8375a5b6b

MODIFIED *  configure include/version/version.m4 bz_ops/k_small_symmetry_group.F hamiltonian/Apply_W_field.F

Bugs:
-

Additions:
 -Apply symmetries to J and P
  and small symmetry group to Wk

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Just a comment

Version 5.3.0,  Revision 23946,  Hash 4c632eb53e

MODIFIED *  configure hamiltonian/Apply_W_field.F

Bugs:
-

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23947,  Hash ecfcf600c3

MODIFIED *  configure include/version/version.m4 hamiltonian/Berry_polarization_NEQ.F nloptics/NL_Berry_current.F

Bugs:
 -Fixed i_time_rev in symmetries number

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23948,  Hash 8e0a89b2d

MODIFIED *  include/version/version.m4 bz_ops/.objects

DELETED *  bz_ops/k_small_symmetry_group.F

Bugs:
-

Additions:
-

Changes:
 -Removed small group

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23949,  Hash 0ce90d876

MODIFIED *  include/version/version.m4 hamiltonian/Apply_W_field.F modules/mod_R_lattice.F modules/mod_electric.F nloptics/NL_Hamiltonian.F nloptics/NL_initialize.F sc/SC_add_efield.F

Bugs:
-

Additions:
-

Changes:
 -Removed all the small group part

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23950,  Hash 3294a88bd

MODIFIED *  include/version/version.m4 modules/mod_nl_optics.F

Bugs:
 -Missing deallocation of Sigma_sex_eq

Additions:
 -Added new operator Wk_EQ

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23951,  Hash 51525ee19

MODIFIED *  include/version/version.m4 modules/mod_nl_optics.F nloptics/NL_Hamiltonian.F nloptics/NL_driver.F

Bugs:
-

Additions:
 -Prepare code for symmetrization

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23952,  Hash cda916933

MODIFIED *  include/version/version.m4 nloptics/NL_Hamiltonian.F nloptics/NL_driver.F

Bugs:
-

Additions:
-

Changes:
 -Prepared code for the symmetrization of DIP_iR

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23953,  Hash 48396a662b

MODIFIED *  nloptics/NL_Hamiltonian.F

Bugs:
-

Additions:
 -Implemented W(E)*E=W(0)*E+[W(E)-W(0)]*E

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23954,  Hash c27f4142b

MODIFIED *  include/version/version.m4

NEW *  io/io_full_SYMMs.F

Bugs:
-

Additions:
 -Added new subroutine to store the full symmetries

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23955,  Hash 0f02196bf

MODIFIED *  include/version/version.m4 io/.objects

Bugs:
-

Additions:
 -New file to store the full symmetries of the system before removal

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23956,  Hash 515446f81

MODIFIED *  include/version/version.m4 io/io_full_SYMMs.F

Bugs:
-

Additions:
 -Fixed typos

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23957,  Hash e8f604c0f7

MODIFIED *  nloptics/NL_Hamiltonian.F ypp/symmetries/symmetries_driver.F

Bugs:
-

Additions:
 -Added storing old symmetries

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23958,  Hash 366c96f508

MODIFIED *  io/io_full_SYMMs.F modules/mod_D_lattice.F ypp/symmetries/symmetries_driver.F

Bugs:
-

Additions:
-

Changes:
 -Old symmetries put in a module

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23959,  Hash 11f506d091

MODIFIED *  include/version/version.m4 io/io_full_SYMMs.F modules/mod_D_lattice.F ypp/symmetries/symmetries_driver.F

Bugs:
-

Additions:
 -it compiles

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23960,  Hash ab2b9bf75e

MODIFIED *  include/version/version.m4 nloptics/.objects nloptics/NL_driver.F ypp/symmetries/symmetries_driver.F

DELETED *  config/stamps_and_lists/project_dependencies.stamp

Bugs:
-

Additions:
 -Started symmetrization procedure

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23961,  Hash 3440bf6da

MODIFIED *  include/version/version.m4 bz_ops/DOUBLE_project.dep hamiltonian/NL_project.dep io/DOUBLE_project.dep ypp/symmetries/symmetries_driver.F

Bugs:
-

Additions:
 -Fixed typos

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23962,  Hash 45f3a34a8

MODIFIED *  include/version/version.m4 nloptics/NL_driver.F

NEW *  nloptics/DIP_symmetrize.F

Bugs:
-

Additions:
 -Added DIPOLE symmetrize

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23963,  Hash dd4b33cdb

MODIFIED *  include/version/version.m4

RENAMED *  nloptics/DIP_symmetrize.F -> nloptics/DIPOLE_symmetrize.F

Bugs:
-

Additions:
 -Added missing file

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23964,  Hash 3f297e5d7

MODIFIED *  include/version/version.m4 nloptics/DIPOLE_symmetrize.F

Bugs:
-

Additions:
 -Added dipole symmetrization with full symmetries

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Merge branch 'tech-master' into nl-symmetries

attacc committed:
Version 5.3.0,  Revision 23968,  Hash 8e8f00702

MODIFIED *  configure include/version/version.m4

Bugs:
-

Additions:
-

Changes:
 -Regerated configure

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23969,  Hash 6eb834288

MODIFIED *  configure include/version/version.m4 nloptics/DIPOLE_symmetrize.F nloptics/DOUBLE_project.dep nloptics/NL_project.dep

Bugs:
-

Additions:
-Fixed symmetrization

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23970,  Hash 52da1b449

MODIFIED *  configure include/version/version.m4 io/io_OSCLL.F io/io_full_SYMMs.F nloptics/DIPOLE_symmetrize.F

Bugs:
-

Additions:
 -Better IO

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23971,  Hash 1fbcd112f

MODIFIED *  configure include/version/version.m4 io/io_full_SYMMs.F nloptics/DIPOLE_symmetrize.F

Bugs:
-

Additions:
 -Work in progress

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23972,  Hash 4e826bb44

MODIFIED *  configure include/version/version.m4 io/io_full_SYMMs.F modules/mod_D_lattice.F nloptics/DIPOLE_symmetrize.F ypp/symmetries/fix_ATOMPROJs.F ypp/symmetries/fix_PPs.F ypp/symmetries/fix_WFs_and_E.F ypp/symmetries/fix_symmetries.F ypp/symmetries/symmetries_driver.F

Bugs:
-

Additions:
 -It works

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Merge branch 'tech-master' into nl-symmetries

attacc committed:
Merge branch 'tech-master' into nl-symmetries

attacc committed:
Version 5.3.0,  Revision 23978,  Hash 80e4b1b33

MODIFIED *  configure include/version/version.m4 nloptics/DIPOLE_symmetrize.F

Bugs:
 -Fixed two bugs

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23979,  Hash f9b476093

MODIFIED *  configure include/version/version.m4 nloptics/DIPOLE_symmetrize.F

Bugs:
-

Additions:
 -It works!!!!!

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23980,  Hash a0c51b22a

MODIFIED *  configure include/version/version.m4 bz_ops/k_map_nearest_by_dir.F dipoles/DIPOLE_overlaps.F

Bugs:
-

Additions:
-

Changes:
 -Change notation in WF_shifts according to Davide suggestion WF_shifts=-1,0,1

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23981,  Hash 770bedc37

MODIFIED *  configure include/version/version.m4 hamiltonian/Berry_polarization_NEQ.F interface/INIT_activate.F interface/INIT_load.F modules/mod_electric.F nloptics/NL_Berry_current.F

Bugs:
-

Additions:
 -Reversed use of residual symmetries

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23982,  Hash ffa591cbe

MODIFIED *  configure include/version/version.m4 nloptics/NL_initialize.F

Bugs:
 -Fixed typos

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23983,  Hash f66f30b47

MODIFIED *  configure include/version/version.m4 hamiltonian/.objects hamiltonian/DOUBLE_project.dep hamiltonian/NL_project.dep hamiltonian/RT_project.dep hamiltonian/SC_project.dep nloptics/NL_Hamiltonian.F real_time_hamiltonian/.objects real_time_hamiltonian/DOUBLE_project.dep real_time_hamiltonian/RT_Hamiltonian.F real_time_hamiltonian/RT_Hamiltonian_diagonalize.F real_time_hamiltonian/RT_project.dep

RENAMED *  real_time_hamiltonian/RT_apply_field.F -> hamiltonian/Apply_field.F

Bugs:
-

Additions:
 -Apply field moved to Hamiltonian

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23984,  Hash d275a2c55

MODIFIED *  configure include/version/version.m4 nloptics/NL_Hamiltonian.F

Bugs:
 -Fixed typos

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23985,  Hash 1278d044c

MODIFIED *  configure include/version/version.m4 nloptics/NL_driver.F

Bugs:
-

Additions:
-Fixed velocity gauge case

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23986,  Hash a02f928ac

MODIFIED *  configure include/version/version.m4 nloptics/DIPOLE_symmetrize.F

Bugs:
-

Additions:
 -Just a temporary error message

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23987,  Hash 961880401

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_W_operator.F hamiltonian/Build_tilde_vbands.F

Bugs:
-

Additions:
 -Better coding

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23988,  Hash b6c3a1703

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_W_operator.F

Bugs:
-

Additions:
 -Test in progress

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23989,  Hash e6eb7550d

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_W_operator.F

Bugs:
-

Additions:
-

Changes:
 -Test in progress

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23990,  Hash c9be52f5a

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_W_operator.F hamiltonian/Build_tilde_vbands.F nloptics/NL_Berry_current.F

Bugs:
-

Additions:
 -Better coding

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23991,  Hash 78050e87c

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_W_operator.F

Bugs:
-

Additions:
 -Better code

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23992,  Hash 896fa31b6

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_W_operator.F

Bugs:
-

Additions:
-Better code

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23993,  Hash 8115f924d

MODIFIED *  configure include/version/version.m4 nloptics/DIPOLE_symmetrize.F

Bugs:
 -Fixed parallelization

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 23994,  Hash 4461255

MODIFIED *  modified: modified: modified:

Bugs:
 -Fixed W operator

Additions:
-

Changes:
-

Patch sent by:  attacc <claudio.attaccalite@gmail.com>

attacc committed:
Merge branch 'tech-master' into nl-symmetries

attacc committed:
Version 5.3.0,  Revision 24089,  Hash 4f2dd7b99

MODIFIED *  configure include/version/version.m4 nloptics/DIPOLE_symmetrize.F

Bugs:
-

Additions:
 -Removed comment

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24090,  Hash 13726b3d8

MODIFIED *  configure include/version/version.m4 modules/mod_Overlaps.F nloptics/DIPOLE_symmetrize.F

Bugs:
 -Bug in definition of io_Overlaps

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24091,  Hash cef35ac8b

MODIFIED *  configure include/version/version.m4 modules/mod_Overlaps.F

Bugs:
-

Additions:
 -Wrong interface to io_Overlaps

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24092,  Hash 457edd74f

MODIFIED *  configure include/version/version.m4 nloptics/DIPOLE_symmetrize.F

Bugs:
-

Additions:
 -Work in progress

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24093,  Hash f7a178a2c

MODIFIED *  configure include/version/version.m4 nloptics/DIPOLE_symmetrize.F

Bugs:
-

Additions:
 -Work in progress

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Merge branch 'fix_iusse_195' into nl-symmetries

attacc committed:
Merge branch 'fix_iusse_195' into nl-symmetries

attacc committed:
Version 5.3.0,  Revision 24098,  Hash ac08273b8

MODIFIED *  configure include/version/version.m4 bse/DOUBLE_project.dep hamiltonian/Build_W_operator.F

Bugs:
-

Additions:
 -Code optimized with DeepSeek for cgemm substitution

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24099,  Hash fcd633652

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_tilde_vbands.F

Bugs:
-

Additions:
 -BLAS optimization with DeepSeek

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24100,  Hash 49a64820b

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_tilde_vbands.F

Bugs:
-

Additions:
-

Changes:
 -small changes

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Merge remote-tracking branch 'origin/tech-master' into nl-symmetries

attacc committed:
Version 5.3.0,  Revision 24104,  Hash 443163f0c

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_tilde_vbands.F

Bugs:
-

Additions:
 -Fixed typos

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24105,  Hash 74d76d380

MODIFIED *  configure include/version/version.m4 interface/INIT_activate.F interface/INIT_load.F nloptics/NL_Berry_current.F nloptics/NL_driver.F nloptics/NL_initialize.F

Bugs:
-

Additions:
 -Current to second order in dk

Changes:
 -Better flag for symmetries

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24106,  Hash 1af0418fe

MODIFIED *  configure include/version/version.m4 nloptics/NL_output.F

Bugs:
 -Fixed output angle number

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24107,  Hash ce2259a60

MODIFIED *  configure include/version/version.m4 interface/INIT_load.F io/variables_NL.F modules/mod_nl_optics.F nloptics/NL_initialize.F

Bugs:
-

Additions:
 -Added possibility to select max angle

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24108,  Hash 178d42f4e

MODIFIED *  configure include/version/version.m4 nloptics/DIPOLE_symmetrize.F nloptics/NL_initialize.F

Bugs:
-

Additions:
 -Max angle is plotted in output

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24109,  Hash 77fb8bca2

MODIFIED *  configure include/version/version.m4 modules/SET_defaults.F

Bugs:
-

Additions:
 -Fixed default

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24110,  Hash 152fe9d

MODIFIED *  modified: modified: modified:

Bugs:
-

Additions:
 -Forgot variable

Changes:
-

Patch sent by:  attacc <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24111,  Hash dd4214047

MODIFIED *  configure include/version/version.m4 hamiltonian/Apply_field.F nloptics/NL_driver.F

Bugs:
-

Additions:
 -Added Doxygen documentation

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24112,  Hash 01fd274e1

MODIFIED *  configure include/version/version.m4 hamiltonian/Apply_W_field.F

Bugs:
 -Fixed in/out for a variable

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

sangallidavide commented:
Two points about the "residual symmetries".

1) In principle it is a good idea when computing equilibrium quantities, such as the dipoles.
However, it is tricky and it might add some phases mismatches w.r.t. the wave-functions if used in the wrong way. Maybe you don't, but I provide an example just to clarify what I mean.

Suppose you have a SAVE with full symetries, with k3, k4 and k5 both outside the IBZ and reached by symmetry operations S3, S4, S5 from k2.
Then you remove some symmetries, and now k3 enters in the IBZ with reduced symmetries, k4 remains in the star of k2 reached by S4, while k5 enters in the star of k3, reached via S2. Accordingly, in the SAVE with reduced symmetries yambo will then use S3*S2 to reach k5, while is will use S5 in the SAVE with all symmetries. This introduces a phase.

2) While the "residual symmetries" are useful for equilibrium quantities, they should not be used for NEQ quantities, since the action of the external field lift these symmetries. From a quick look it seems that the "residual symmetries" are used for NEQ polarization and current.

attacc commented:
Hi Davide

about point 

1) I think dipole is well defined because is <psi_k | r| psi_k >  so any additional phase is cancel out in the bra-ket. However I checked that linear and non-linear response are identical after resymmetrization. If I force symmetry in the dipole I just get a slightly better result in the sense that elements that should be zero are more close to zero then without re-symmetrization

about the point 
2) when I speak of residual symmetries I mean symmetry that are not removed by ypp, therefore is legitimate that I apply them in the dynamics (residual symmetris are not the full symmetries of the system)

attacc commented:
Anyway these are not important changes... I can also remove some part from this pull request

andreamarini commented:
Hi all. This PR touches some of the routines I am heavily coding at the moment. 

It will take time for me to complete the generalized Floquet approach.

If merged I would not be able to take care of the conflicts. In a  separate plugin/module/project, instead, it would be perfect and we could test it and discuss it in person to see what can be exported to the core.

:smile:

attacc commented:
This pull request is very very small ...... at maximum just move a file from RT to Hamiltonian
you cannot handle this? too complicated?

attacc commented:
If there is a part that you do not like just let me know

sangallidavide reviewed (approved):
Thanks Claudio for clarifying about syms.

For me the Pr is ok and can be merged.

attacc committed:
Version 5.3.0,  Revision 24113,  Hash 2531c6adf

MODIFIED *  configure include/version/version.m4 nloptics/NL_Hamiltonian.F nloptics/NL_driver.F

Bugs:
-

Additions:
 -Better code, use less memory

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24114,  Hash 26fcb49b4

MODIFIED *  configure include/version/version.m4 hamiltonian/.objects nloptics/NL_Hamiltonian.F real_time_hamiltonian/.objects real_time_hamiltonian/RT_Hamiltonian_diagonalize.F

RENAMED *  hamiltonian/Apply_field.F -> real_time_hamiltonian/RT_apply_field.F

Bugs:
-

Additions:
-

Changes:
 -Restored RT_apply_field.F

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24115,  Hash 207a032f9

MODIFIED *  configure include/version/version.m4 hamiltonian/.objects hamiltonian/DOUBLE_project.dep hamiltonian/NL_project.dep hamiltonian/RT_project.dep hamiltonian/SC_project.dep interface/INIT_activate.F interface/INIT_load.F nloptics/NL_Hamiltonian.F nloptics/NL_driver.F real_time_hamiltonian/DOUBLE_project.dep real_time_hamiltonian/RT_Hamiltonian.F real_time_hamiltonian/RT_Hamiltonian_diagonalize.F real_time_hamiltonian/RT_project.dep

Bugs:
-

Additions:
 -Added V_parallel verbosity for threads
 -RT_apply_field restored

Changes:
 -QP correction in yambo_nl moved from G to K

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24116,  Hash be65b9296

MODIFIED *  configure include/version/version.m4 io/variables_NL.F

Bugs:
-

Additions:
 -Fixed compatibility with old DBs

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24117,  Hash 3e05c447b

MODIFIED *  configure include/version/version.m4 interface/INIT_activate.F nloptics/NL_driver.F qp_control/QP_apply_interpolate_QP.F

Bugs:
-

Additions:
 -Fixed io QP_interpoalted in yambo_nl

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24118,  Hash aaced4753

MODIFIED *  configure include/version/version.m4 io_parallel/io_COLLISIONS_header.F

Bugs:
 -Fixed bug with collisions compression that was always zero

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc commented:
I restored the RT_apply_field file

attacc committed:
Version 5.3.0,  Revision 24119,  Hash bfc16a43c

MODIFIED *  configure include/version/version.m4 io/io_header.F qp_control/QP_apply_interpolate_QP.F

Bugs:
-

Additions:
 -Better change in QP_apply_interpoalte_QP.F

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24120,  Hash 33213d7ab

MODIFIED *  configure include/version/version.m4 io/io_header.F

Bugs:
-

Additions:
-

Changes:
 -Forgot comment

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc commented:
@sangallidavide please check if this change  in QP_apply_interpolated_QP.F is better for you.
Now only the master writes. The point is that in yambo_nl all processors write databases.

attacc committed:
Merge remote-tracking branch 'origin/5.3' into nl-symmetries

attacc committed:
Merge branch 'tech-master' into nl-symmetries

attacc committed:
Version 5.3.0,  Revision 24128,  Hash 48fa31a33

MODIFIED *  configure include/version/version.m4

Bugs:
-

Additions:
 -Configure updated

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24129,  Hash cf3d3ede0

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_tilde_vbands.F hamiltonian/NL_project.dep hamiltonian/RT_project.dep

Bugs:
-

Additions:
-

Changes:
 -Code simplification

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24119,  Hash bfc16a4

MODIFIED *  modified: modified: modified: modified: modified:

Bugs:
-

Additions:
-

Changes:
 -The code does not search for collisions if Correlation==IPA

Patch sent by:  attacc <claudio.attaccalite@gmail.com>

attacc committed:
Merge branch 'nl-symmetries' of github.com:yambo-code/yambo into nl-symmetries

attacc committed:
Version 5.3.0,  Revision 24132,  Hash 75371e8

MODIFIED *  modified: modified: modified:

Bugs:
 -Fixed bug with intel-fortran compiler

Additions:
-

Changes:
-

Patch sent by:  attacc <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24133,  Hash 9b3179e

MODIFIED *  modified: modified:

Bugs:
-

Additions:
-

Changes:
 -Configure updated

Patch sent by:  attacc <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24134,  Hash 01c254c

MODIFIED *  modified: modified: modified:

Bugs:
 -Missing flag in NL input

Additions:
-

Changes:
-

Patch sent by:  attacc <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24135,  Hash d0acaaa

MODIFIED *  modified: modified: modified:

Bugs:
 -Fixed incompatibility with ifort

Additions:
-

Changes:
-

Patch sent by:  attacc <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24136,  Hash 07eeed955

MODIFIED *  configure include/version/version.m4 nloptics/NL_driver.F

Bugs:
 -Fixed bug introduced in previous commit with RPA/TDDFT collisions

Additions:
-

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24137,  Hash a2a0f7a42

MODIFIED *  configure include/version/version.m4

Bugs:
-

Additions:
-

Changes:
 -configure regenated

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24134,  Hash 01c254cb3

MODIFIED *  configure include/version/version.m4 qp_control/QP_apply_interpolate_QP.F

Bugs:
-

Additions:
-

Changes:
 -Removed _NL flags according to Davide suggestion

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Merge branch 'nl-symmetries' of github.com:yambo-code/yambo into nl-symmetries

attacc committed:
Version 5.3.0,  Revision 24140,  Hash bf78d1c

MODIFIED *  modified: modified: modified:

Bugs:
 -Fixed bug with IFORT again

Additions:
-

Changes:
-

Patch sent by:  attacc <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24141,  Hash c769dccf1

MODIFIED *  configure include/version/version.m4 hamiltonian/Build_tilde_vbands.F xc_functionals/DOUBLE_project.dep xc_functionals/RT_project.dep

Bugs:
-

Additions:
 -Restored old version without cgemm (postponed for later)

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24142,  Hash 23b46697d

MODIFIED *  configure include/version/version.m4 collisions/COLLISIONS_HXC.F xc_functionals/DOUBLE_project.dep xc_functionals/RT_project.dep

Bugs:
-

Additions:
-

Changes:
 -Small change in collisions SC to avoid crash on my machine

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc committed:
Version 5.3.0,  Revision 24143,  Hash 77ed66f02

MODIFIED *  configure include/version/version.m4 collisions/COLLISIONS_HXC.F modules/mod_collision_ext.F

Bugs:
-

Additions:
 -Restored old file and initialized more variables

Changes:
-

Patch sent by:  Claudio Attaccalite <claudio.attaccalite@gmail.com>

attacc commented:
I added tests on Farah
https://media.yambo-code.eu/robots/GPL/nl-symmetries/mo.farah.1.php

to be compared with tech-master ones
https://media.yambo-code.eu/robots/GPL/tech-master/mo.farah.5.php

no new errors are present in nl-symmetries
ciao
cla

sangallidavide reviewed (approved):
Fine with me.

The only point on which I'm not 100% sure is about the dipoles symmetrization, in case the order in which symmetries are applied are not consistent.

attacc commented:
Probably I can remove that part, I tested it on different systems, not
great improvements....
results are more or less the same.
I think it is a bit useless..

cla

On Thu, 27 Mar 2025 at 14:49, Davide Sangalli ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> Fine with me.
>
> The only point on which I'm not 100% sure is about the dipoles
> symmetrization, in case the order in which symmetries are applied are not
> consistent.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/yambo-code/yambo/pull/199#pullrequestreview-2721606305>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AED6PCPDH6FPP27CLAZWLDD2WPXWBAVCNFSM6AAAAABY5Q4U6KVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDOMRRGYYDMMZQGU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
==========================================================
Mobile[FR]:    +33662922897
Mobile[IT]:     +393384315437
Fax:              +33491418916
Address: CINaM
            Campus de Luminy
            163 Avenue de Luminy, Case 913
            13288 Marseille Cedex 9

Skype: claudioattaccalite
Editor Fellow for: www.scipost.org
web site: www.attaccalite.com
===========================================================
